lib/marshalingCommands.js

```diff
@@ -0,0 +1,173 @@
+var _ = require('lodash');
+var unexpectedItemHandler = require('./unexpectedItemHandler');
+
+/**
+ * Converts boolean value to DynamoDb "BOOL"
+ *
+ * @param item
+ * @returns {*}
+ */
+function marshalBoolean(item) {
+  if (!_.isBoolean(item)) {
```

Comment:
I might be misunderstanding you, but after playing with your idea a bit I don't think it makes the code any more clear or DRY.

```js
var _ = require('lodash');
var isNotBoolean = _.negate(_.isBoolean);

function test(predicate, item) {
  return predicate(item) ? item : void 0;
}

function marshalBoolean(item) {
  var testedItem = test(isNotBoolean, item);
  
  if (testedItem) {
    return { BOOL: testedItem };
  }
}
```

I'm not going to do this one. While this is a fine approach it feels much less clear than the way it is currently and doesn't really make an impact on DRYness either.